src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs

```diff
@@ -80,8 +80,12 @@ public void Handle(CommandLineArguments added, CommandLineArguments removed, IWo
 
             if (diff.AddedItems.Count > 0 || diff.RenamedItems.Count > 0 || diff.ChangedItems.Count > 0)
             {
-                // Make sure the tree matches the same version of the evaluation that we're handling
-                IProjectTreeServiceState treeState = await PublishTreeAsync(e).ConfigureAwait(true); // TODO: https://github.com/dotnet/roslyn-project-system/issues/353
+                // Make sure the tree matches the same version of the evaluation that we're handling for active project context.
+                // If we are dealing with non- active project context, then just use the latest published tree.
+                // TODO: https://github.com/dotnet/roslyn-project-system/issues/353
+                IProjectTreeServiceState treeState = isActiveContext ?
```

Comment:
Sure, let me also file an issue so your suggested fix can be implemented in future.